File: personalizatio-sdk/src/main/java/com/personalizatio/stories/models/elements/ButtonElement.java
package com.personalizatio.stories.models.elements;

import androidx.annotation.NonNull;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Objects;

public class ButtonElement implements LinkElement {

    private String title;
    private String background;
    private String color;
    private boolean textBold = false;
    private final String link;

    public ButtonElement(@NonNull JSONObject json) throws JSONException {
        if (json.has("title")) {
            title = json.getString("title");
        }
        if (json.has("background")) {
            background = json.getString("background");
        }
        if (json.has("color")) {
            color = json.getString("color");
        }
        if (json.has("text_bold")) {
            textBold = json.getBoolean("text_bold");
        }
        link = getLink(json);
    }

    public String getTitle() {
        return title;
    }

    public String getBackground() {
        return background;
    }

    public String getColor() {
        return color;
    }

    public Boolean getTextBold() {
        return textBold;
    }

    @Override
    public String getLink() {
        return link;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ButtonElement that)) return false;
        return textBold == that.textBold
                && title.equals(that.title)
                && background.equals(that.background)
                && color.equals(that.color)
                && link.equals(that.link);
    }

    @Override
    public int hashCode() {
        return Objects.hash(title, background, color, textBold, link);
    }

    @NonNull
    @Override
    public String toString() {
        return "ButtonElement{" +
                "title='" + title + '\'' +
                ", background='" + background + '\'' +
                ", color='" + color + '\'' +
                ", textBold=" + textBold +
                ", link='" + link + '\'' +
                '}';
    }
}

File: personalization-sdk/src/main/kotlin/com/personalization/stories/models/elements/ButtonElement.kt
package com.personalization.stories.models.elements

import org.json.JSONObject
import java.util.Objects

class ButtonElement(json: JSONObject) : LinkElement {
    val title: String = json.optString("title", "")
    val background: String = json.optString("background", DEFAULT_BACKGROUND)
    val color: String = json.optString("color", DEFAULT_COLOR)
    val textBold: Boolean = json.optBoolean("text_bold", false)
    override val link: String = getLink(json)

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ButtonElement) return false
        return textBold == other.textBold
                && title == other.title
                && background == other.background
                && color == other.color
                && link == other.link
    }

    override fun hashCode(): Int {
        return Objects.hash(title, background, color, textBold, link)
    }

    override fun toString(): String {
        return "ButtonElement{title='$title', background='$background', color='$color', textBold=$textBold, link='$link'}"
    }

    companion object {
        private const val DEFAULT_BACKGROUND = "#FFFFFF"
        private const val DEFAULT_COLOR = "#000000"
    }
}
